### DISCUSSION TITLE
Pass state to `installation.*` webhook events

### DISCUSSION THREAD

**Discussion by ajhenry** [community/community#34608] [Apps, API and Webhooks]
Currently, you can pass `state` to [an installation via a URL param](https://docs.github.com/en/developers/apps/managing-github-apps/installing-github-apps#preserving-an-application-state-during-installation). But this state can only be accessed via a setup URL from fetching the query param.

It would be awesome if the state was in the payload of the webhook that gets called for `installation.*` events. This would probably eliminate the need for a setup URL in the majority of workflows. Additionally, this would make testing significantly easier.

**Comment by shakedbuk57** [comment]
was this ever solved?
